Helm Chart for 5.0.5

pjuarezd opened the pull request:
Helm chart 5.0.5

Changes in this release:
- Converted bandwidth Pod to Job for one-time test execution
- Converted latency Pod to Job with post-install hook for automated testing
- Updated Chart version to 5.0.5
- Enhanced StatefulSet configuration with improved resource management
- Added new configuration options in values.yaml for test parameters
- Fixed service port configuration to use integer type

These changes provide better Kubernetes-native handling of test workloads,
ensuring proper completion tracking and automatic cleanup of finished tests.

pjuarezd committed:
Helm chart 5.0.5: Convert Pods to Jobs for improved orchestration

Changes in this release:
- Converted bandwidth Pod to Job for one-time test execution
- Converted latency Pod to Job with post-install hook for automated testing
- Updated Chart version to 5.0.5
- Enhanced StatefulSet configuration with improved resource management
- Added new configuration options in values.yaml for test parameters
- Fixed service port configuration to use integer type

These changes provide better Kubernetes-native handling of test workloads,
ensuring proper completion tracking and automatic cleanup of finished tests.

pjuarezd committed:
add helm hook to create bandwidth pod after hperf statefulset

pjuarezd committed:
update notes

ramondeklein commented on helm/hperf/templates/service.yaml:
```diff
@@ -8,7 +8,7 @@ spec:
   clusterIP: None  
   ports:
-    - port: 9999
+    - port: {{ .Values.hperf.port }}
```
This will change the default port from `9999` to `9010`. Not sure if backwards compatibility is a requirement.

pjuarezd replied:
new hperf 5.0.5 default port is 9010

```
./hperf server
starting 'hperf' server on: 0.0.0.0:9010
```

ramondeklein commented on helm/hperf/templates/statefulset.yaml:
```diff
@@ -5,6 +5,7 @@ metadata:
   namespace: {{ .Release.Namespace | quote }}
   labels:
     app: hperf
+    {{- include "hperf.labels" . | nindent 4 }}
```
Would be good to include `hperf.labels: {}` in the `values.yaml` file.

ramondeklein commented on helm/hperf/templates/statefulset.yaml:
```diff
@@ -32,12 +33,19 @@ spec:
-            - name: http1
-              containerPort: 9999
+            - name: server
+              containerPort: {{ .Values.hperf.port | default 9010}}
```
No need for two defaults, so please remove the `| default 9010` (it's already specified in `values.yaml`).

ramondeklein commented on helm/hperf/templates/statefulset.yaml:
```diff
@@ -56,3 +64,24 @@ spec:
+          type: {{ .Values.hperf.hperfVolume.hostPath.type | default "DirectoryOrCreate" }}
+            {{- else if eq .Values.hperf.volumeType "persistentVolumeClaim" }}
+        # Default to emptyDir if no volume type specified
+        emptyDir: { }
```
This will only be generated when `volumeType: persistentVolumeClaim`. It's not the fallback... I also don't think this `emptyDir` will map to the volume claim template on line 84.

pjuarezd replied:
right, fixed on https://github.com/minio/hperf/pull/24/commits/bc8b9766a2427e2e4865b1db07a9cc5895d220da

ramondeklein commented on helm/hperf/templates/latency-job.yaml:
```diff
@@ -0,0 +1,68 @@
+    {{- include "hperf.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "1"
```
Same here.

pjuarezd replied:
removed

ramondeklein commented on helm/hperf/templates/bandwith-job.yaml:
```diff
@@ -0,0 +1,68 @@
+    {{- include "hperf.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "1"
```
Why does this need a `post-install` hook? After the creation of the statefulset, it isn't available immediately. It needs to rely on retries anyway.

pjuarezd replied:
without the hook the Job pod is created before the servers are ready, and that prints a errors of server not available, I much rather preffer to not have that kind of distractions and only start the Job pods after the statefulset is avalable.

```
hperf-bandwidth-xqthd 
hperf-bandwidth-xqthd  Test ID: bandwidth-test
hperf-bandwidth-xqthd 
hperf-bandwidth-xqthd ERROR:  dial tcp: lookup hperf-3.hperf.hperf.svc.cluster.local on 10.96.0.10:53: no such host
hperf-bandwidth-xqthd ERROR:  dial tcp: lookup hperf-0.hperf.hperf.svc.cluster.local on 10.96.0.10:53: no such host
hperf-bandwidth-xqthd ERROR:  dial tcp: lookup hperf-5.hperf.hperf.svc.cluster.local on 10.96.0.10:53: no such host
hperf-bandwidth-xqthd ERROR:  dial tcp: lookup hperf-4.hperf.hperf.svc.cluster.local on 10.96.0.10:53: no such host
hperf-bandwidth-xqthd ERROR:  dial tcp: lookup hperf-1.hperf.hperf.svc.cluster.local on 10.96.0.10:53: no such host
hperf-bandwidth-xqthd ERROR:  dial tcp: lookup hperf-2.hperf.hperf.svc.cluster.local on 10.96.0.10:53: no such host
hperf-bandwidth-xqthd ERROR:  dial tcp: lookup hperf-4.hperf.hperf.svc.cluster.local on 10.96.0.10:53: no such host
hperf-bandwidth-xqthd ERROR:  dial tcp: lookup hperf-3.hperf.hperf.svc.cluster.local on 10.96.0.10:53: no such host
hperf-bandwidth-xqthd ERROR:  dial tcp: lookup hperf-2.hperf.hperf.svc.cluster.local on 10.96.0.10:53: no such host
hperf-bandwidth-xqthd #ERR   #TX        TX(high)   TX(low)    TX(total)       #Dropped  Mem(high) Mem(low)  CPU(high) CPU(low)  
hperf-bandwidth-xqthd 0      420        11.58 GB/s 1.36 GB/s  130.11 GB       0         3         2         3         0         
hperf-bandwidth-xqthd 0      780        11.61 GB/s 1.36 GB/s  245.60 GB       0         3         2         67        0         
hperf-bandwidth-xqthd 0      1140       11.79 GB/s 1.36 GB/s  360.29 GB       0         3         2         69        0         
hperf-bandwidth-xqthd 0      1440       11.79 GB/s 1.36 GB/s  455.52 GB       0         3         2         69        0         
hperf-bandwidth-xqthd 0      1860       11.79 GB/s 1.36 GB/s  588.79 GB       0         3         2         69        0         
hperf-bandwidth-xqthd 0      2220       11.79 GB/s 1.36 GB/s  703.78 GB       0         3         2         69        0         
hperf-bandwidth-xqthd 0      2520       11.79 GB/s 1.36 GB/s  799.18 GB       0         3         2         69        0         
hperf-bandwidth-xqthd 0      2880       11.79 GB/s 1.35 GB/s  913.62 GB       0         3         2         69        0         

But with the hook errors are printed anyways, I think is the service DNS that takes 30 seconds to propagate, oh well I am removing the hooks

pjuarezd committed:
add labels field

pjuarezd committed:
remove hooks

pjuarezd committed:
remove default

pjuarezd committed:
mount volume based on options: emptyDir, hostPath or volumeClaimTemplate

pjuarezd committed:
Set print all as default to true

pjuarezd committed:
Comment latency test by default and only print NOTES.txt instructions when latency is enabled

pjuarezd committed:
Re-run the Job OnFailure

If the test fails because the hperf servers were not ready before the Job started, we should allow the container test to be recreated.

pjuarezd committed:
Bind IP to the server pods
By binding the actual pod IP and not the `0.0.0.0` interface we can get mode detailed results of test
